test/connect.js

```diff
@@ -416,3 +419,245 @@ test('websocket clients have access to the request object', function (t) {
+test('tcp proxied (protocol v1) clients have access to the ipAddress property', function (t) {
+  t.plan(1)
+
+  var port = 1885
+  var packet = {
+    cmd: 'connect',
+    protocolId: 'MQTT',
+    protocolVersion: 4,
+    clean: true,
+    clientId: 'my-client-proxyV1',
+    keepalive: 0
+  }
+
+  var src = new proxyProtocol.Peer('127.0.0.1', 12345)
+  var dst = new proxyProtocol.Peer('127.0.0.1', port)
+  var protocol = new proxyProtocol.V1BinaryProxyProtocol(
+    proxyProtocol.INETProtocol.TCP4,
+    src,
+    dst,
+    mqttPacket.generate(packet)
+  ).build()
+
+  var broker = aedes({
+    preConnect: function (client, done) {
+      client.ip = client.ipAddress
+      return done()
+    },
+    trustProxy: true
+  })
+
+  var server = net
+    .createServer(broker.handle)
+    .listen(port, function (err) {
+      t.error(err, 'no error')
+    })
+
+  broker.on('client', function (client) {
+    if (client.ip) {
+      t.pass('ip address present')
+      t.equal('127.0.0.1', client.ip)
+      finish()
+    } else {
+      t.fail('no ip address present')
+    }
+  })
+
+  var client = net.createConnection(
+    {
+      port,
+      // host: parsedProto.destination.ipAddress,
+      // localAddress: parsedProto.source.ipAddress,
+      // localPort: parsedProto.source.port,
+      timeout: 250
+    }
+  )
+
+  client.on('timeout', function () {
+    client.write(protocol)
+  })
+
+  var timer = setTimeout(finish, 1000)
+
+  function finish () {
+    clearTimeout(timer)
+    client.end()
+    broker.close()
+    server.close()
+    t.end()
+  }
+})
+
+// mocking mqtt packet delivered by a server using proxy protocol v2
+test('tcp proxied (protocol v2) clients have access to the ipAddress property', function (t) {
+  t.plan(3)
+
+  var port = 1883
+  var packet = {
+    cmd: 'connect',
+    protocolId: 'MQTT',
+    protocolVersion: 4,
+    clean: true,
+    clientId: 'my-client-proxyV2',
+    keepalive: 0
+  }
+
+  var protocol = new proxyProtocol.V2ProxyProtocol(
+    proxyProtocol.Command.LOCAL,
+    proxyProtocol.TransportProtocol.DGRAM,
+    new proxyProtocol.IPv4ProxyAddress(
+      proxyProtocol.IPv4Address.createFrom([127, 0, 0, 1]),
+      12345,
+      proxyProtocol.IPv4Address.createFrom([127, 0, 0, 1]),
+      1883
+    ),
+    mqttPacket.generate(packet)
+  ).build()
+
+  var broker = aedes({
+    preConnect: function (client, done) {
+      client.ip = client.ipAddress
+      return done(null, true)
+    },
+    trustProxy: true
+  })
+
+  var server = net
+    .createServer(broker.handle)
+    .listen(port, function (err) {
+      t.error(err, 'no error')
+    })
+
+  broker.on('client', function (client) {
+    if (client.ip) {
+      t.pass('ip address present')
+      t.equal('127.0.0.1', client.ip)
+      finish()
+    } else {
+      t.fail('no ip address present')
+    }
+  })
+
+  var client = net.createConnection(
+    {
+      port,
+      // host: parsedProto.proxyAddress.destinationAddress.address.join('.),
```

Comment:
why is this commented out?

Reply:
It was intended to show how to manipulate the object. It's now into the example so i'll remove it.